packages/adaptive_scaffold/lib/src/adaptive_layout.dart

```diff
@@ -0,0 +1,509 @@
+  ///
+  /// Note: if using flexibly sized Widgets like [Container], wrap the Widget in a
+  /// [SizedBox] or limit its size by another method.
+  final SlotLayout? primaryNavigation;
+
+  /// The slot placed on the end side of the screen.
+  ///
+  /// Note: if using flexibly sized Widgets like [Container], wrap the Widget in a
+  /// [SizedBox] or limit its size by another method.
+  final SlotLayout? secondaryNavigation;
+
+  /// The slot placed on the top part of the screen.
+  ///
+  /// Note: if using flexibly sized Widgets like [Container], wrap the Widget in a
+  /// [SizedBox] or limit its size by another method.
+  final SlotLayout? topNavigation;
+
+  /// The slot placed on the bottom part of the screen.
+  ///
+  /// Note: if using flexibly sized Widgets like [Container], wrap the Widget in a
+  /// [SizedBox] or limit its size by another method.
+  final SlotLayout? bottomNavigation;
+
+  /// The slot that fills the rest of the space in the center.
+  final SlotLayout? body;
+
+  /// A supporting slot for [body]. Has a sliding entrance animation by default.
+  /// The default ratio for the split between [body] and [secondaryBody] is so
+  /// that the split axis is in the center of the screen.
+  final SlotLayout? secondaryBody;
+
+  /// Defines the fractional ratio of [body] to the [secondaryBody].
+  ///
+  /// For example 0.3 would mean [body] takes up 30% of the available space
+  /// and[secondaryBody] takes up the rest.
+  ///
+  /// If this value is null, the ratio is defined so that the split axis is in
+  /// the center of the screen.
+  final double? bodyRatio;
+
+  /// Whether or not the developer wants the smooth entering slide transition on
+  /// [secondaryBody].
+  ///
+  /// Defaults to true.
+  final bool internalAnimations;
+
+  /// Whether to orient the body and secondaryBody in horizontal order (true) or
+  /// in vertical order (false).
+  ///
+  /// Defaults to true.
+  final bool horizontalBody;
+
+  @override
+  State<AdaptiveLayout> createState() => _AdaptiveLayoutState();
+}
+
+class _AdaptiveLayoutState extends State<AdaptiveLayout>
+    with TickerProviderStateMixin {
+  late AnimationController _controller;
+
+  late Map<String, SlotLayoutConfig?> chosenWidgets =
+      <String, SlotLayoutConfig?>{};
+  Map<String, Size?> slotSizes = <String, Size?>{
+    _kPrimaryNavigationID: Size.zero,
+    _kSecondaryNavigationID: Size.zero,
+    _kTopNavigationID: Size.zero,
+    _kBottomNavigationID: Size.zero,
+  };
+
+  Map<String, ValueNotifier<Key?>> notifiers = <String, ValueNotifier<Key?>>{
+    _kPrimaryNavigationID: ValueNotifier<Key?>(null),
+    _kSecondaryNavigationID: ValueNotifier<Key?>(null),
+    _kTopNavigationID: ValueNotifier<Key?>(null),
+    _kBottomNavigationID: ValueNotifier<Key?>(null),
+    _kBodyID: ValueNotifier<Key?>(null),
+    _kSecondaryBodyID: ValueNotifier<Key?>(null),
+  };
+
+  Map<String, bool> isAnimating = <String, bool>{
+    _kPrimaryNavigationID: false,
```

Comment:
I think that instead of a `Map<String, bool>` you should just use a `Set<String>`, and if it's in the set, it's assumed to be `true`.